adapters/repos/db/vector/hnsw/insert.go

```diff
@@ -105,6 +108,111 @@ func (h *hnsw) AddBatch(ctx context.Context, ids []uint64, vectors [][]float32)
 	return nil
 }
 
+func (h *hnsw) AddMultiBatch(ctx context.Context, docIDs []uint64, vectors [][][]float32) error {
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+	if !h.multivector.Load() {
+		return errors.Errorf("AddMultiBatch called on non-multivector index")
+	}
+	if len(docIDs) != len(vectors) {
+		return errors.Errorf("ids and vectors sizes does not match")
+	}
+	if len(docIDs) == 0 {
+		return errors.Errorf("insertBatch called with empty lists")
+	}
+
+	h.trackDimensionsOnce.Do(func() {
+		atomic.StoreInt32(&h.dims, int32(len(vectors[0][0])))
```

Comment:
Add checks that `vectors[0][0]` exists

Reply:
Yes

Reply:
Could we not have different dimensions in multi-vector?